src/Workspaces/CSharp/Portable/LanguageServices/CSharpSemanticFactsService.cs

```diff
@@ -371,5 +371,11 @@ private SymbolInfo GetSymbolInfo(SemanticModel semanticModel, SyntaxNode node, S
 
         public bool IsInsideNameOfExpression(SemanticModel semanticModel, SyntaxNode node, CancellationToken cancellationToken)
             => (node as ExpressionSyntax).IsInsideNameOfExpression(semanticModel, cancellationToken);
+
+        public bool CanConvert(SemanticModel semanticModel, SyntaxNode node, ITypeSymbol type)
+        {
+            var conversion = semanticModel.ClassifyConversion((ExpressionSyntax)node, type) ;
+            return conversion.Exists && conversion.IsImplicit;
+        }
```

Comment:
i have to look.  but i could have sworn the IDE had a way to do this in a language agnostic manner.

Reply:
I'll try. I'm concerned that might not work with typeless expressions like `null` or `default` though.

Reply:
It's somewhat strange we don't have a ClassifyCommonConversion that takes an expression.  Could that be added instead?

Reply:
I suspect that's what `SemanticModel.ClassifyConversion` is for.

Reply:
Let me rephrase.  I'm surprised there is no ClassifyCommonConversion on SemanticModel.  Can that be added instead?

Reply:
Ah, makes sense. I didn't know the "Common" nomenclature for cross-language APIs.

Reply:
it was actually not desirable.  but it was the best we could come up with so that we wouldn't conflict with the existign language-specific apis :)